core/src/test/java/io/atomix/core/AtomixTest.java

```diff
@@ -246,6 +250,152 @@ public void testRoleChangedListener() throws Exception {
     roleChanged.get();
   }
 
+  @Test
+  public void testAtomixBootstrapPartitions() throws Exception {
+    // given
+    // Partitions \ Nodes
+    //      \   0  1  2
+    //    0     L  F  F
+    //    1     F  L  F
+    //    2     F  F  L
+
+    final List<Map<Integer, RaftServer.Role>> nodeRoles = new ArrayList<>();
+    nodeRoles.add(new HashMap<>());
+    nodeRoles.add(new HashMap<>());
+    nodeRoles.add(new HashMap<>());
+
+    final List<Integer> members = Arrays.asList(1, 2, 3);
+
+    final int firstNodeId = 1;
+    final CompletableFuture<Atomix> nodeOneFuture = setupAtomixWithListener(firstNodeId, members, nodeRoles);
+
+    final int secondNodeId = 2;
+    final CompletableFuture<Atomix> nodeTwoFuture = setupAtomixWithListener(secondNodeId, members, nodeRoles);
+
+    final int thirdNodeId = 3;
+    final CompletableFuture<Atomix> nodeThreeFuture = setupAtomixWithListener(thirdNodeId, members, nodeRoles);
+
+    // then
+    CompletableFuture.allOf(nodeOneFuture, nodeTwoFuture, nodeThreeFuture).join();
+
+    while (nodeRoles.get(0).size() < 3
```

Comment:
Can we add a time out in case these tests fail? :sweat_smile: 

You can use a copy of the one I added in the Raft suite (`ZeebeTestHelper`):

```java
public void await(final BooleanSupplier predicate) {
    final long tries = Duration.ofMillis(DEFAULT_TIMEOUT_MS).toNanos() / 100;
    boolean result = predicate.getAsBoolean();
    for (long i = 0; i < tries && !result; i++) {
      LockSupport.parkNanos(10_000);
      result = predicate.getAsBoolean();
    }

    assertTrue(result);
  }
```